stl/inc/optional

```diff
@@ -53,6 +53,13 @@ protected:
     _THROW(bad_optional_access{});
 }
 
+struct _Nontrivial_dummy_type {

```

Comment:
I've been able to follow why this is type is nontrivial, but I don't actually see any references to this new type in the file (other than its definition).  Could someone provide some intuition around how this is actually used in `optional`?

Reply:
I think @CaseyCarter has accidentally reverted most of the change in the last commit

Reply:
![image](https://user-images.githubusercontent.com/456873/84197812-6749fc00-aa57-11ea-9021-eb4d901fecd9.png)

Ouch - I backed out the `char` to `_Nontrivial_dummy_type` change to capture the generated code for the "middle" condition and failed to put it before pushing.

Reply:
(Maybe I'm the dummy and not this type,)